plugins/rest_connector/api/tasks.py

```diff
@@ -113,3 +120,24 @@ def perform_request(self, connector_id: str, db_id: int) -> str:
     )
 
     return "finished"
+
+
+def _download_files(
+    request_file_descriptors: List[Dict],
+) -> Dict[str, Tuple[str, BytesIO, str]]:
+    # TODO: cache downloaded files for subsequent requests?
+    request_files = {}
+
+    for i, req_file in enumerate(request_file_descriptors):
+        file_url = req_file["source"]
+        response = requests.get(file_url)
```

Comment:
we may be able to use the response directly as a file like object if we open the request in streaming mode, but this will be more robust